app/src/main/java/com/eveningoutpost/dexdrip/InfluxDB/InfluxDBUploader.java

```diff
@@ -0,0 +1,146 @@
+package com.eveningoutpost.dexdrip.InfluxDB;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+
+import com.eveningoutpost.dexdrip.Models.BgReading;
+import com.eveningoutpost.dexdrip.Models.Calibration;
+import com.eveningoutpost.dexdrip.Models.UserError.Log;
+
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDBFactory;
+import org.influxdb.dto.BatchPoints;
+import org.influxdb.dto.Point;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import okhttp3.HttpUrl;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Response;
+
+public class InfluxDBUploader {
+    private static final int SOCKET_TIMEOUT = 60000;
+    private static final int CONNECTION_TIMEOUT = 30000;
+    private static final String TAG = InfluxDBUploader.class.getSimpleName();
+    private SharedPreferences prefs;
+    private Context mContext;
+    private String dbName;
+    private String dbUri;
+    private String dbUser;
+    private String dbPassword;
+    private OkHttpClient.Builder client;
+
+    public InfluxDBUploader(Context context) {
+        mContext = context;
+        prefs = PreferenceManager.getDefaultSharedPreferences(mContext);
+        dbName = prefs.getString("cloud_storage_influxdb_database", null);
+        dbUri  = prefs.getString("cloud_storage_influxdb_uri", null);
+        dbUser = prefs.getString("cloud_storage_influxdb_username", null);
+        dbPassword = prefs.getString("cloud_storage_influxdb_password", null);
+
+        client = new OkHttpClient.Builder()
+                .connectTimeout(CONNECTION_TIMEOUT, TimeUnit.MILLISECONDS)
+                .readTimeout(SOCKET_TIMEOUT, TimeUnit.MILLISECONDS)
+                .writeTimeout(SOCKET_TIMEOUT, TimeUnit.MILLISECONDS)
+                .addNetworkInterceptor(new Interceptor() {
+                    @Override
+                    public Response intercept(Chain chain) throws IOException {
+                        HttpUrl url = HttpUrl.parse(dbUri);
+                        String fullPath = url.encodedPath() + chain.request().url().encodedPath();
```

Comment:
To get this to work locally, I had to filter on a double leading //

`String fullPath = (url.encodedPath() + chain.request().url().encodedPath()).replaceFirst("^//","/");`